### DISCUSSION TITLE
Samples you're looking for, but can't find?

### DISCUSSION THREAD

**Discussion by jmatthiesen** [dotnet/ai-samples#64] [Q&A]
We've provided an initial set of samples for working with AI in .NET, but there are a LOT of possibilities in this space. Are there any specific AI samples you've been looking for in .NET?

**Comment by TheBlueSky** [comment]
I want to know how to use `Microsoft.Extensions.AI` with OpenAI compatible API, such as OpenRouter, Chutes, or even Ollama.
When I tried to use it with no success.

With OpenRouter I got:
```
Error: System.Text.Json.JsonReaderException: '<' is an invalid start of a value. LineNumber: 0 | BytePositionInLine: 0.
```

With Chutes I got:
```
Error: System.ClientModel.ClientResultException: Service request failed.
Status: 404 (Not Found)
```

I don't know what the issue is with OpenRouter, but with Chutes it looks like the library adds `/api` to the endpoint base URL, which Chutes does not have; their URL is `https://llm.chutes.ai/v1/chat/completions`.
